core/src/network/peer_manager.cpp

```diff
@@ -462,4 +462,16 @@ void PeerManager::updateConnections() {
     }
 }
 
+std::vector<libp2p::peer::PeerId> PeerManager::getPeersInfo() const {
+    std::vector<libp2p::peer::PeerId> peers;
+    auto key_selector = [](auto pair){return pair.first;};
```

Comment:
You may potentially want fully connected peers only, so you could consider filtering by state